management/src/main/java/org/kaazing/gateway/management/monitoring/configuration/impl/MonitorFileWriterImpl.java

```diff
@@ -0,0 +1,316 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.kaazing.gateway.management.monitoring.configuration.impl;
+
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.MappedByteBuffer;
+
+import org.kaazing.gateway.management.monitoring.configuration.MonitorFileWriter;
+import org.kaazing.gateway.management.monitoring.service.MonitoredService;
+import org.kaazing.gateway.management.monitoring.writer.GatewayWriter;
+import org.kaazing.gateway.management.monitoring.writer.ServiceWriter;
+import org.kaazing.gateway.management.monitoring.writer.impl.MMFGatewayWriter;
+import org.kaazing.gateway.management.monitoring.writer.impl.MMFSeviceWriter;
+import org.kaazing.gateway.service.MonitoringEntityFactory;
+
+import uk.co.real_logic.agrona.BitUtil;
+import uk.co.real_logic.agrona.DirectBuffer;
+import uk.co.real_logic.agrona.concurrent.UnsafeBuffer;
+
+/**
+ * Class responsible with storing/writing information regarding the MMF format.
+ *
+ * File layout:
+ * +-----------------------------------------------------------------------+
+ * | File version | GW data offset | Service mappings offset | | GW ID | GW counters
+ * lbl buffer offset | GW counters lbl buffer length | GW counters values buffer
+ * offset | GW counters values buffer length | | Number of services | Service 1 name
+ * | Service 1 offset | ... | Service 1 lbl buffer offset | Service 1 lbl buffer length |
+ * | Service 1 values buffer offset | Service 1 values buffer length | | ... |
+ * | GW counters labels buffer | | GW counters values buffer | | Service 1 labels buffer |
+ * | Service 1 values buffer || ... |
+ * +-----------------------------------------------------------------------+
+ * Metadata length: NUMBER_OF_INTS_IN_HEADER * BitUtil.SIZE_OF_INT + SIZEOF_STRING +
+ * servicesCount * (SIZEOF_STRING + NUMBER_OF_INTS_PER_SERVICE * BitUtil.SIZE_OF_INT)
+ */
+public final class MonitorFileWriterImpl implements MonitorFileWriter {
+    private static final int OFFSETS_PER_SERVICE = 4;
+    private static final int NUMBER_OF_INTS_PER_SERVICE = 5;
+    private static final int NUMBER_OF_INTS_IN_HEADER = 8;
+    private static final int SIZEOF_STRING = 128;
+    private static final int MAX_SERVICE_COUNT = 10;
```

Comment:
10 services is a bit limiting. I would recommend 100, or set it based on the actual number of services plus some margin (like 10).